lib/phoenix_live_view/lifecycle.ex

```diff
@@ -106,15 +106,12 @@ defmodule Phoenix.LiveView.Lifecycle do
   end
 
   @doc false
-  def on_mount(view, view), do: raise_own_mount_hook!(view, view)
-  def on_mount(view, {view, :mount} = id), do: raise_own_mount_hook!(view, id)
-
-  def on_mount(_view, {module, fun} = id) when is_atom(module) and is_atom(fun) do
-    hook!(id, :mount, Function.capture(module, fun, 3))
+  def on_mount(_view, {module, arg}) when is_atom(module) do
+    mount_hook!({module, arg})
   end
 
   def on_mount(_view, module) when is_atom(module) do
-    hook!(module, :mount, Function.capture(module, :on_mount, 3))
+    mount_hook!({module, :default})
```

Comment:
@josevalim I went with `:default` for the arg à la [`Broadway.handle_message/3`](https://hexdocs.pm/broadway/Broadway.html#c:handle_message/3), wdyt?

Reply:
Sounds good to me!